docs/tutorial/part04.rst

```diff
@@ -2,5 +2,437 @@
+execute the following python import commands::
+
+    from fetch_api.models import Entity
+    from fetch_api.models import Intermediary
+    from fetch_api.models import Officer
+    from fetch_api.models import Address
+    from fetch_api.models import Other
+
+Each of the models we just imported maps to a specific structure of a node label, property keys and
+relationship types in the PPGDB. Now we are actually ready to start exploring the
+Paradise Paper Graph Database through the Neomodel Query API.
+
+..  todo::
+    Put screenshot image of the console here.
+
+Neomodel Query API
+==================
+Each of your models has some properties and methods(inherited from StructureNode or DjangoNode)
+that help us to express queries to the Neo4j database.
+
+NodeSet and Nodes Neomodel objects
+--------------------------------------------------------
+A *NodeSet* object represents a set of nodes matching common query parameters or filters.
+
+A *Node* object is an instance of one of our models. That means we can access all the properties
+and methods defined on the model class. Each instance represents a single node in the database.
+
+The ``<Model>.nodes`` class property of each model store a NodeSet object. Each time we access
+this ``.nodes`` property we get a brand new nodeset object, which means we get nodeset without any
+filters applied. Initially, before applying any filters, this noseset represents all the nodes
+mapped under a model(nodes labeled with the same class name). For instance, ``Entity.nodes``
+contains all the nodes with the label Entity on the database.
+
+Later we will see how we can apply filters in order to match a specific subset of nodes.
+
+Length of a NodeSet
+-------------------
+If we wanted to count all the Entity nodes that are stored in the database, we just call
+the ``len`` python function over the ``Entity.nodes`` nodeset.
+
+Example::
+
+    len(Entity.nodes)
+
+When we call ``len(Entity.nodes)``, Neomodel will generate a cypher query that counts
+all the nodes with the label ``Entity``. Then that query is executed in the Neo4j database and
+we get back the count. The cypher query string that is generated by Neomodel behind the scene is::
+
+    MATCH (n:Entity) RETURN COUNT(n)
+
+.. note::
+    We are not retrieving all the nodes from the database and then count them. The actual counting
+    is done by the Neo4j database engine which is faster.
+
+Another example, to get count of all the nodes that exist in the PPGDB database::
+
+    len(Entity.nodes) \
+    + len(Officer.nodes) \
+    + len(Intermediary.nodes) \
+    + len(Address.nodes) \
+    + len(Other.nodes)
+
+..  todo::
+    Put screenshot image of the console here.
+
+If nodeset is filtered, only nodes that fulfill the filters will be counted.
+
+Fetching nodes
+----------------
+In order to retrieve the nodes, read their properties and relationships, an actual cypher query need

```

Comment:
At the end of the line; can we change `need` for `needs`

Reply:
http://www.phrasemix.com/phrases/something-needs-to-be-done 👍